viz_scripts/auxiliary_files/energy_intensity.csv

```diff
@@ -1,17 +1,15 @@
-﻿replaced_mode,EI
-"Car, drove alone",1.54100
-Taxi/Uber/Lyft,2.441
-Bus,1.3300
-Train,0.2300
-Free Shuttle,1.3300
-"Car, with others",0.7700
-Bikeshare,0.0220
-Scooter share,0.0270
-Pilot ebike,0.0220
-Walk,0.0000
-Skate board,0.0000
-Regular Bike,0.0000
-Not a Trip,0.0000
-No Travel,0.0000
-Same Mode,0.0000
-Other,nan
+﻿mode,fuel,(kWH)/PMT,(kWH)/trip,EI(kWH/PMT-trip),energy_intensity_factor,energy_intensity_units,CO2_factor,CO2_factor_units
+"Car, drove alone",gasoline,1.541,0,1.541,5170,BTU/PMT,157.2,lb_CO2/MMBTU
+"Car, with others",gasoline,0.77,0,0.77,2585,BTU/PMT,157.2,lb_CO2/MMBTU
+Taxi/Uber/Lyft,gasoline,2.441,0,2.441,7214,BTU/PMT,157.2,lb_CO2/MMBTU

```

Comment:
we don't actually use the (kWH)/PMT values for these, right, we use the BTU and multiply. So you could just leave them blank (which will be `N/A` in the dataframe).

Because I am not sure that these numbers are correct (5170 * 0.000293071) = 1.515, not 1.541 (unless I am missing something). That's why I like to let the computer do the calculations to the extent possible instead of relying on us humans.

Reply:
You're correct the (kWH)/PM is not use in the calculation. I left that column there to see where the other values where coming from. But I just updated the energy_intensity.csv as you suggested.

This is also correct (5170 * 0.000293071) = 1.515, not 1.541. Probably a typo on my end.

Reply:
The energy intensity per trip is handle by energy_intensity fucntion.  Do you want the comment in the actual notebook cell where I execute this fucntion, or in the scaffolding.py ?

The implementation was not as hard once I combine the BUT/PMT and the kWh/PMT in one column. I was trying to keep them in separate because of the units. Also, as you can see the impact functions had a bit of repeat code because I'm reading both replaced_mode and mode_confirm into the fucntion, and I did not how to repeat the code to apply to both column and store the variables separate and use it them in final calculation (subtraction = impact). Anyway, if none of these make sense. I can explain later, but if you have tips/suggestions/ways to make my code prettier and dryer. I'll be happy to take a lesson or two. :)

Reply:
I'm a bit confused. the `energy_intensity` method  only uses the `energy_intensity_factor`, but I don't see how you can calculate that without looking at the number of trips. For example, in the table below, how did you get `0.0311` from 0.027 and 0.0041? I assumed you would actually have a per trip calculation that was based on the distance, and a per dataset calculation that was based on the number of trips.

If you could clarify where you are doing that, it would be great.


mode | fuel | (kWH)/PMT | (kWH)/trip | EI(kWH/PMT-trip) | energy_intensity_factor |
-- | -- | -- | -- | -- | -- |
Scooter share | electric | 0.027 | 0.0041 | 0.0311 | 0.0311

Reply:
I don't think that works. Let us consider a trip distance of 10 miles.

If you use the expanded formula, kWh = 10 * 0.027 + 0.0041 = 0.27 + 0.0041 = 0.2741
If you use your simplified formula, kWh = 10 * 0.0311 = 0.311

The problem is that the 0.0041 is also multiplied by the distance when it should not be.

Reply:
ok, I see the mistake.

I think I fixed it . I changed where the ei_kWH/trip was added, please the a look and let me know.